src/dataregistry/registrar/base_table_class.py

```diff
@@ -87,3 +100,38 @@ def modify(self, entry_id, modify_fields):
         """
 
         raise NotImplementedError
+
+    def find_entry(self, entry_id):
+        """
+        Find an entry in the database.
+
+        Parameters
+        ----------
+        entry_id : int
+            Unique identifier for table entry
+            e.g., dataset_id for the dataset table
+
+        Returns
+        -------
+        r : CursorResult object
+            Found entry (None if no entry found)
+        """
+
+        # Search for dataset in the registry.
+        my_table = self._get_table_metadata(self.which_table)
+
+        if self.which_table == "dataset":
+            stmt = select(my_table).where(my_table.c.dataset_id == entry_id)
+        else:
+            raise ValueError("Can only perform `find_entry` on dataset table for now")
```

Comment:
This is ok for now, but we should get rid of this restriction.
For example, add arguments `table_name`, `primary_key`  `to BaseTable.__init__( )` and save those values as `self.table_name`, `self.primary_key`.  (Or maybe one or both of these could be looked up using sqlalchemy functions.) Then `find_entry` works for all tables.